buffer.go

```diff
@@ -517,6 +573,206 @@ func (b *buffer) Float64Column(name string, val float64) *buffer {
 	return b
 }
 
+func (b *buffer) Float64ColumnBinary(name string, val float64) *buffer {
+	if !b.prepareForField() {
+		return b
+	}
+	b.lastErr = b.writeColumnName(name)
+	if b.lastErr != nil {
+		return b
+	}
+	b.WriteByte('=')
+	// binary format flag
+	b.WriteByte('=')
+	b.WriteByte(byte(float64Code))
+	if isLittleEndian {
+		b.Write((*[8]byte)(unsafe.Pointer(&val))[:])
+	} else {
+		data := make([]byte, 8)
+		binary.LittleEndian.PutUint64(data, math.Float64bits(val))
+		b.Write(data)
+	}
+	b.hasFields = true
+	return b
+}
+
+func (b *buffer) Float64Array1DColumn(name string, values []float64) *buffer {
+	if !b.prepareForField() {
+		return b
+	}
+	b.lastErr = b.writeColumnName(name)
+	if b.lastErr != nil {
+		return b
+	}
+	if values == nil {
+		b.writeNullArray()
+		return b
+	}
+
+	dim1 := len(values)
+	if dim1 > MaxArrayElements {
+		b.lastErr = fmt.Errorf("array size %d exceeds maximum limit %d", dim1, MaxArrayElements)
+		return b
+	}
+	b.writeFloat64ArrayHeader(1)
+
+	// Write shape
+	b.writeInt32(int32(dim1))
+
+	// Write values
+	if len(values) > 0 {
+		b.writeFloat64Data(values)
+	}
+
+	b.hasFields = true
+	return b
+}
+
+func (b *buffer) Float64Array2DColumn(name string, values [][]float64) *buffer {
+	if !b.prepareForField() {
+		return b
+	}
+	b.lastErr = b.writeColumnName(name)
+	if b.lastErr != nil {
+		return b
+	}
+
+	if values == nil {
+		b.writeNullArray()
+		return b
+	}
+
+	// Validate array shape
+	dim1 := len(values)
+	var dim2 int
+	if dim1 > 0 {
+		dim2 = len(values[0])
+		totalElements := dim1 * dim2
+		if dim1 > MaxArrayElements || dim2 > MaxArrayElements || totalElements > MaxArrayElements || totalElements < 0 {
```

Comment:
We need to improve overflow detection:
dim1 = 65536, dim2 = 65536
totalElements = 65536 * 65536 = 4294967296
On 32-bit systems: overflows to 0, bypasses validation